src/transaction.ts

```diff
@@ -1219,8 +1227,16 @@ export class Snapshot extends EventEmitter {
           this._update(response.metadata!.transaction);
         }
       })
-      .on('error', () => {
-        if (!this.id && this._useInRunner) {
+      .on('error', err => {
```

Comment:
For a different refactor PR: It seems that `runStream` and `createReadStream` have a lot of duplicated code. Could we combine more of those?

Reply:
Sure.